### PR TITLE
Remove "mutable" from LSPTypechecker GlobalState pointer

### PR DESCRIPTION
**Pull request by aprocter**
This removes the `mutable` qualifier from the `GlobalState` pointer in `LSPTypechecker`. In order to do this, we have to change `GlobalState::replaceFile` so that it doesn't "borrow" a unique_ptr to the GlobalState.

### Motivation
Just seems nice to drop `mutable` in cases where we don't actually need it.

### Test plan
Existing CI.

- [x] Some manual smoke testing against Stripe codebase with LSP

### CODE REVIEW

File: main/lsp/LSPTypechecker.h [hunk 1]
```diff
@@ -34,9 +34,8 @@ class UndoState;
 class LSPTypechecker final {
     /** Contains the ID of the thread responsible for typechecking. */
     std::thread::id typecheckerThreadId;
-    /** GlobalState used for typechecking. Mutable because typechecking routines, even when not changing the GlobalState
-     * instance, actively consume and replace GlobalState. */
-    mutable std::unique_ptr<core::GlobalState> gs;
```
Comment (froydnj) [inline main/lsp/LSPTypechecker.h:39]: This change is one of those times that I wish `this->` prefixing was pervasive, so it was easier to see at a glance where members were getting mutated versus parameters.

**Review by froydnj — APPROVED** [review]
Makes sense to me!  Have you tested this on Stripe's codebase with vscode at all, just to make sure LSP isn't going to blow up somehow in real-world usage?

**Comment by aprocter** [comment]
> Have you tested this on Stripe's codebase with vscode at all, just to make sure LSP isn't going to blow up somehow in real-world usage?

Nope! Will be doing that before merge.

**Comment by aprocter-stripe** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_L06oXDIPJWGrSr
→ https://go/builds/bui_L06oqTwTwPs6gN

**Pull request merged**